src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
@ctrueden The reason I wanted to stick with `Exception` (besides following the `CacheLoader` interface) was to be able to notify the user about the failure. There might be other, maybe better ways, like a callback that notifies the user on failure.
I also removed the `Functor` interface and replaced it with `CacheLoader` as @axtimwalde suggested, which makes more sense because `IntervalKeyLoaderAsLongKeyLoader` (renamed the class as well) is a Wrapper around another `CacheLoader` (which `throws Exception`). There is three ways of getting rid of `throws Exception` in `IntervalKeyLoaderAsLongKeyLoader`:
 1. Add `NonThrowingCacheLoader< K, V > extends CacheLoader< K, V >` that does not throw and use that as the wrapped loader in `IntervalKeyLoaderAsLongKeyLoader`.
 2. Remove `throws Exception` from `CacheLoader`
 3. Use `try {} catch {}` and just print the exception/notify caller through callback

Out of all these options, `1` is most reasonable to me as it delegates exception handling to the caller who is the only one who knows about what kind of exceptions can occur.